### ISSUE TITLE
Token Usage with LiteLLM

### ISSUE THREAD

**Issue by Yonom** [vercel/ai#2763]
### Description

Using `@ai-sdk/openai` with LiteLLM (custom `baseUrl`) does not automatically send the `stream_options: { token_usage: true }` flag. This option seems to be only sent when `compatibility: "strict"` mode is enabled.

Enabling `compatibility: "strict"` causes the parsing to crash because in some chunks, `completion_tokens` seems to be missing:

```json
{
  "id":"chat-<snip>",
  "choices":[{"index":0,"delta":{"content":"","role":"assistant"}}],
  "created":1724276300,
  "model":"<snip>",
  "object":"chat.completion.chunk",
  "usage":{"prompt_tokens":11,"total_tokens":11}
}
```

causing the following error:

```json
Type validation failed: Value: {"id":"chat-<snip>","choices":[{"index":0,"delta":{"content":"","role":"assistant"}}],"created":1724276300,"model":"<snip>","object":"chat.completion.chunk","usage":{"prompt_tokens":11,"total_tokens":11}}.
Error message: [
  {
    "code": "invalid_union",
    "unionErrors": [
      {
        "issues": [
          {
            "code": "invalid_type",
            "expected": "number",
            "received": "undefined",
            "path": [
              "usage",
              "completion_tokens"
            ],
            "message": "Required"
          }
        ],
        "name": "ZodError"
      },
      {
        "issues": [
          {
            "code": "invalid_type",
            "expected": "object",
            "received": "undefined",
            "path": [
              "error"
            ],
            "message": "Required"
          }
        ],
        "name": "ZodError"
      }
    ],
    "path": [],
    "message": "Invalid input"
  }
]
```

Expectation: completion_tokens should be parsed as 0 if its omitted. There should be _some_ way to track token usage with litellm

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/2775

**Comment by lgrammel** [comment]
Fixed in `@ai-sdk/openai@0.0.52`

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete parsing failure in existing OpenAI-compatible streaming behavior, and the maintainer confirmed it was fixed in a package release.

### Evidence
- With a custom LiteLLM base URL, strict compatibility streaming crashes when a usage chunk omits `completion_tokens`.
- The reporter expected omitted `completion_tokens` to be treated as zero so token usage remains available.
- A maintainer linked PR #2775 and confirmed the issue was fixed in `@ai-sdk/openai@0.0.52`.
